File: src/Frames.java
package src;

class Frame {
    private String frameType;
    private boolean tierA1; // True means it fits on machine
    private boolean tierA2;
    private boolean tierB; // False means it does not fit on machine

    public Frame() {
        frameType = null;
        tierA1 = false;
        tierA2 = false;
        tierB = false;
    }

    // Getters for frameType, tierA1, tierA2, & tierB
    public String getFrameType() {
        return frameType;
    }
    public boolean getTierA1() {
        return tierA1;
    }
    public boolean getTierA2() {
        return tierA2;
    }
    public boolean getTierB() {
        return tierB;
    }

    // Setters for frameType, tierA1, tierA2, & tierB
    public void setFrameType(String frametype) {
        this.frameType = frametype;
    }
    public void setTierA1(boolean tier_A1) {
        this.tierA1 = tier_A1;
    }
    public void setTierA2(boolean tier_A2) {
        this.tierA2 = tier_A2;
    }
    public void setTierB(boolean tier_B) {
        this.tierB = tier_B;
    }
}

public class Frames {
    public static void main(String[] args) {

    }
}

File: src/Frames.py
class Frame:
    def __init__(self):
        self.name = None
        self.tierA1 = 0  # 0 means that by default no machine was assigned tier A1 for this frame
        self.tierA2 = 0
        self.tierB = 0  
    
    # Getters for frame name and machine for tiers A1, A2, & B
    def get_name(self):
        return self.name
    
    def get_tierA1(self):
        return self.tierA1
    
    def get_tierA2(self):
        return self.tierA2
    
    def get_tierB(self):
        return self.tierB

    # Setters for frame name and machine for tiers A1, A2, & B
    def set_name(self, frame_name):
        self.name = frame_name
    
    def set_tierA1(self, machine):
        self.tierA1 = machine

    def set_tierA2(self, machine):
        self.tierA2 = machine

    def set_tierB(self, machine):
        self.tierB = machine
